common/persistence/dataInterfaces.go

```diff
@@ -1477,6 +1489,8 @@ type (
 
 		// Remove Task due to corrupted data
 		DeleteTask(request *DeleteTaskRequest) error
+
+		ListExecutions(request *ListExecutionsRequest) (*ListExecutionsResponse, error)
```

Comment:
how do you plan to use this API in the mutable state scanner ? Will the scanner first call this api to get a list of (wfID,runID) and then make another call to GetWorkflowExecution() for each one ? In Cassandra, executions table is weird in that it is multiple tables in one. Some entries will have "special" runIDs which have no meaning.

Reply:
1. How will this be used: yes two calls was my idea. Its possible for this API to return more information but not all persistence layers can do this with a single table scan, so I am returning Ids so that all persistence stores can be used the same way. 

2. When I run the scan I filter based on row type ```query := d.session.Query(templateListExecutionsQuery, d.shardID, rowTypeExecution)```. I understood this to me that I will only . get meaningful runIds. Do you know if this is not the case?

Reply:
ah.. i missed the rowTypeExecution part.

Reply:
Let's hold onto this PR until andrew is back from vacation. I want to have a brief sync around the API i.e. returning (wfID, runID) as opposed to more info related to mutable state